core/arch/riscv/kernel/entry.S

```diff
@@ -0,0 +1,233 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2022 NXP
+ */
+
+#include <asm.S>
+#include <generated/asm-defines.h>
+#include <keep.h>
+#include <platform_config.h>
+#include <riscv.h>
+#include <riscv_macros.S>
+#include <mm/core_mmu.h>
+
+.section .data
+.balign 4
+
+#ifdef CFG_BOOT_SYNC_CPU
+.equ SEM_CPU_READY, 1
+#endif
+
+	/*
+	 * Setup sp to point to the top of the tmp stack for the current CPU:
+	 * sp is assigned:
+	 * stack_tmp + cpu_id * stack_tmp_stride - STACK_TMP_GUARD
+	 */
+.macro set_sp
+	/* Unsupported CPU, park it before it breaks something */
+	li	t1, CFG_TEE_CORE_NB_CORE
+	csrr	t0, CSR_XSCRATCH
+	bge	t0, t1, unhandled_cpu
+	la	t2, stack_tmp
+	lw	t1, stack_tmp_stride
+	/*
+	 * t0 is core pos + 1
+	 * t1 is value of stack_tmp_stride
+	 * t2 is value of stack_tmp
+	 */
+	mv	sp, t2
+	mult	t0, t1, t2
+	add	sp, sp, t2
+.endm
+
+.macro cpu_is_ready
+#ifdef CFG_BOOT_SYNC_CPU
+	csrr	t0, CSR_XSCRATCH
+	la	t1, sem_cpu_sync
+	slli	t0, t0, 2
+	add	t1, t1, t0
+	li	t2, SEM_CPU_READY
+	sw	t2, 0(t1)
+	fence
+#endif
+.endm
+
+.macro set_tp
+	li	a1, THREAD_CORE_LOCAL_SIZE
+	la	tp, thread_core_local
+	mult	a0, a1, t0
+	add	tp, tp, t0
+	csrr	a0, CSR_XSCRATCH
+	sw	a0, THREAD_CORE_LOCAL_HART_ID(tp)
+.endm
+
+.macro set_satp
+	la	a1, boot_mmu_config
+	LDR	a0, CORE_MMU_CONFIG_SATP(a1)
```

Comment:
What does this mean? Is LDR a pseudo instruction that is expanded by the assembler? Where is `CORE_MMU_CONFIG_SATP` defined?

Reply:
Defined [here ](https://github.com/OP-TEE/optee_os/blob/master/core/arch/riscv/include/riscv.h#L20-L26) to load a word or doubleword depending on the target.
`CORE_MMU_CONFIG_SATP`  is defined in asm-defines, I'll add the commit to [PR#5701](https://github.com/OP-TEE/optee_os/pull/5701/).